core/src/main/java/com/linecorp/armeria/server/healthcheck/HealthCheckService.java

```diff
@@ -435,24 +435,35 @@ public HttpResponse serve(ServiceRequestContext ctx, HttpRequest req) throws Exc
             if (updateResult != null) {
                 switch (updateResult) {
                     case HEALTHY:
-                        serverHealth.setHealthy(true);
+                        serverHealth.setHealthStatus(HealthStatus.HEALTHY);
```

Comment:
Note: If the default `HealthCheckUpdateHandler` is used, only `HEALTHY` and `UNHEALTHY` will be returned. 

I think users would find it surprising that the server is aware of options such as `DEGRADED`, etc.. but doesn't understand update semantics of such statuses

Reply:
Oh you're right. What do you think of adding an implementation of `HealthCheckUpdateHandler` to handle new status? 🤔

Reply:
Sounds good to me 👍

Reply:
Added `HealthStatusUpdateHandler` 😄